controllers/agentpool_controller_deployment.go

```diff
@@ -224,7 +226,22 @@ func agentPoolDeploymentName(ap *appv1alpha2.AgentPool) string {
 }
 
 func agentPoolPodLabels(ap *appv1alpha2.AgentPool) map[string]string {
-	return map[string]string{
+	label := map[string]string{
 		poolNameLabel: ap.Name,
 	}
+
+	//Attempting to merge
+	if ap.Spec.AgentDeployment != nil && ap.Spec.AgentDeployment.Labels != nil {
+		for key, value := range ap.Spec.AgentDeployment.Labels {
+			label[key] = value
+		}
+	}
+
+	return label
+}
+
+func agentDeploymentAnnotations(ap *appv1alpha2.AgentPool) map[string]string {
```

Comment:
This function looks unfinished because it does not take into account user-defined annotations.